csharp/src/Apache.Arrow.Adbc/Core/AdbcConnection.cs

```diff
@@ -0,0 +1,199 @@
+﻿/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Apache.Arrow.Ipc;
+
+namespace Apache.Arrow.Adbc.Core
+{
+    /// <summary>
+    /// Provides methods for query execution, managing prepared statements, using transactions, and so on.
+    /// </summary>
+    public abstract class AdbcConnection : IDisposable

```

Comment:
I'm totally new to Adbc, so forgive the naïve question.

Can this (and other classes) inherit from `DbConnection`/`DbCommand`/etc so things that work with ADO.NET "just work" with Apache.Arrow.Adbc?

Reply:
What's been done for Go, at least, is to provide a package that implements DbConnection and the other 'standard' DB APIs and wraps any ADBC object underneath, since the motivation is that the 'standard' APIs don't expose Arrow in the first place

Reply:
There are different schools of thought to that. We do have an ADO.NET implementation that sits on top of this. However, ADO.NET is row-oriented, so it felt important to stick to Arrow libraries. We can also publish the ADO.NET one if that’s desired.

Reply:
OK - if the idea/design is to sit an ADO.NET implementation on top of this - then that works for me.